pkg/subctl/resource/schedule_pod.go

```diff
@@ -0,0 +1,186 @@
+/*
+© 2021 Red Hat, Inc. and others.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package resource
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+	"github.com/submariner-io/shipyard/test/e2e/framework"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+type NetworkPodType int
+
+type NetworkPodScheduling int
+
+const (
+	GatewayLabel = "submariner.io/gateway"
+)
+
+const (
+	InvalidScheduling NetworkPodScheduling = iota
+	GatewayNode
+	NonGatewayNode
+)
+
+type NetworkingType bool
+
+const (
+	HostNetworking NetworkingType = true
+	PodNetworking  NetworkingType = false
+)
+
+type PodConfig struct {
+	Name       string
+	ClientSet  *kubernetes.Clientset
+	Scheduling NetworkPodScheduling
+	Networking NetworkingType
+	Namespace  string
+	Command    string
+	Timeout    uint
+}
+
+type NetworkPod struct {
+	Pod       *v1.Pod
+	Config    *PodConfig
+	PodOutput string
+}
+
+func ScheduleNetworkPod(config *PodConfig) (string, error) {
+	if config.Scheduling == InvalidScheduling {
+		config.Scheduling = GatewayNode
+	}
+
+	if config.Namespace == "" {
+		config.Namespace = "default"
+	}
+
+	np := &NetworkPod{Config: config}
+	if err := np.schedulePod(); err != nil {
+		return "", err
+	}
+
+	defer np.deletePod()
+	if err := np.awaitUntilPodFinished(); err != nil {
+		return "", err
+	}
+
+	return np.PodOutput, nil
+}
+
+func (np *NetworkPod) schedulePod() error {
+	networkPod := v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: np.Config.Name,
+			Labels: map[string]string{
+				"app": np.Config.Name,
+			},
+		},
+		Spec: v1.PodSpec{
+			Affinity:      nodeAffinity(np.Config.Scheduling),
+			RestartPolicy: v1.RestartPolicyNever,
+			HostNetwork:   bool(np.Config.Networking),
+			Containers: []v1.Container{
+				{
+					Name:    np.Config.Name,
+					Image:   "quay.io/submariner/nettest:devel",
+					Command: []string{"sh", "-c", "$COMMAND >/dev/termination-log 2>&1 || exit 0"},
+					Env: []v1.EnvVar{
+						{Name: "COMMAND", Value: np.Config.Command},
+					},
+				},
+			},
+			Tolerations: []v1.Toleration{{Operator: v1.TolerationOpExists}},
+		},
+	}
+
+	pc := np.Config.ClientSet.CoreV1().Pods(np.Config.Namespace)
+	var err error
+	np.Pod, err = pc.Create(&networkPod)
+	if err != nil {
+		return errors.WithMessage(err, "error spawning the Pod")
```

Comment:
The caller outputs a similar message.

```suggestion
		return err
```